src/overlays/actors/ovl_En_Ossan/z_en_ossan.c

```diff
@@ -37,227 +114,2413 @@ const ActorInit En_Ossan_InitVars = {
+            return;
+        case CANBUY_RESULT_NEED_RUPEES:
+            EnOssan_SetStateCantGetItem(globalCtx, this, 0x85);
+    }
+}
+
+void EnOssan_HandleCanBuyWeirdEgg(GlobalContext* globalCtx, EnOssan* this) {
+    EnGirlA* item = this->shelfSlots[this->cursorIndex];
+
+    switch (item->canBuyFunc(globalCtx, item)) {
+        case CANBUY_RESULT_SUCCESS_FANFARE:
+            EnOssan_GiveItemWithFanfare(globalCtx, this);
+            this->drawCursor = 0;
+            this->shopItemSelectedTween = 0.0f;
+            item->setOutOfStockFunc(globalCtx, item);
+            return;
+        case CANBUY_RESULT_SUCCESS:
+            item->itemGiveFunc(globalCtx, item);
+            EnOssan_SetStateQuickBuyDialog(globalCtx, this, 0x9A);
+            this->drawCursor = 0;
+            this->shopItemSelectedTween = 0.0f;
+            item->setOutOfStockFunc(globalCtx, item);
+            return;
+        case CANBUY_RESULT_CANT_GET_NOW:
+            func_80078884(NA_SE_SY_ERROR);
+            EnOssan_SetStateCantGetItem(globalCtx, this, 0x9D);
+            return;
+        case CANBUY_RESULT_NEED_RUPEES:
+            func_80078884(NA_SE_SY_ERROR);
+            EnOssan_SetStateCantGetItem(globalCtx, this, 0x85);
+    }
+}
+
+void EnOssan_HandleCanBuyBombs(GlobalContext* globalCtx, EnOssan* this) {
+    EnGirlA* item = this->shelfSlots[this->cursorIndex];
+
+    switch (item->canBuyFunc(globalCtx, item)) {
+        case CANBUY_RESULT_SUCCESS_FANFARE:
+        case CANBUY_RESULT_SUCCESS:
+            item->itemGiveFunc(globalCtx, item);
+            EnOssan_SetStateQuickBuyDialog(globalCtx, this, 0x84);
+            this->drawCursor = 0;
+            this->shopItemSelectedTween = 0.0f;
+            item->setOutOfStockFunc(globalCtx, item);
+            return;
+        case CANBUY_RESULT_CANT_GET_NOW:
+            func_80078884(NA_SE_SY_ERROR);
+            EnOssan_SetStateCantGetItem(globalCtx, this, 0x86);
+            return;
+        case CANBUY_RESULT_NEED_RUPEES:
+            func_80078884(NA_SE_SY_ERROR);
+            EnOssan_SetStateCantGetItem(globalCtx, this, 0x85);
+    }
+}
+
+void EnOssan_BuyGoronCityBombs(GlobalContext* globalCtx, EnOssan* this) {
+    if (LINK_AGE_IN_YEARS == YEARS_CHILD) {
+        if (!(gSaveContext.eventChkInf[2] & 0x20)) {
+            if (gSaveContext.infTable[15] & 0x1000) {
+                EnOssan_SetStateCantGetItem(globalCtx, this, 0x302E);
+            } else {
+                this->stickRightPrompt.isEnabled = false;
+                this->stickLeftPrompt.isEnabled = false;
+                this->drawCursor = 0;
+                this->stateFlag = ENOSSAN_STATE_19;
+            }
+        } else {
+            EnOssan_HandleCanBuyBombs(globalCtx, this);
+        }
+    } else {
+        EnOssan_HandleCanBuyBombs(globalCtx, this);
+    }
+}
+
+void EnOssan_State_ItemSelected(EnOssan* this, GlobalContext* globalCtx, Player* player) {
+    GlobalContext* globalCtx2 = globalCtx; // Necessary for OKs
```

Comment:
```suggestion
void EnOssan_State_ItemSelected(EnOssan* this, GlobalContext* globalCtx2, Player* player) {
    GlobalContext* globalCtx = globalCtx2;
```
This is how we have been doing this.

Reply:
even though the temp is never used?

Reply:
eventually the arg will be changed to `Gamestate* state`, so using the `globalCtx` temp in the function itself will make replacing easier